ark/jsonschema/__tests__/array.test.ts

```diff
@@ -0,0 +1,141 @@
+import { attest, contextualize } from "@ark/attest"
+import { parseJsonSchema } from "@ark/jsonschema"
+
+// TODO: Add compound tests for arrays (e.g. maxItems AND minItems )
+// TODO: Add explicit test for negative length constraint failing (since explicitly mentioned in spec)
+
+contextualize(() => {
+	it("type array", () => {
+		const t = parseJsonSchema({ type: "array" })
+		attest(t.json).snap({ proto: "Array" })
+	})
+
+	it("items & prefixItems", () => {
+		const tItems = parseJsonSchema({ type: "array", items: { type: "string" } })
+		attest(tItems.json).snap({ proto: "Array", sequence: "string" })
+		attest(tItems.allows(["foo"])).equals(true)
+		attest(tItems.allows(["foo", "bar"])).equals(true)
+		attest(tItems.allows(["foo", 3, "bar"])).equals(false)
+
+		const tItemsArr = parseJsonSchema({
+			type: "array",
+			items: [{ type: "string" }, { type: "number" }]
+		})
+		attest(tItemsArr.json).snap({
+			proto: "Array",
+			sequence: { prefix: ["string", "number"] },
+			exactLength: 2
+		})
+		attest(tItemsArr.allows(["foo", 1])).equals(true)
+		attest(tItemsArr.allows([1, "foo"])).equals(false)
+		attest(tItemsArr.allows(["foo", 1, true])).equals(false)
+
+		const tPrefixItems = parseJsonSchema({
+			type: "array",
+			prefixItems: [{ type: "string" }, { type: "number" }]
+		})
+		attest(tPrefixItems.json).snap({
+			proto: "Array",
+			sequence: { prefix: ["string", "number"] },
+			exactLength: 2
+		})
+	})
+
+	it("additionalItems", () => {
+		const tItemsVariadic = parseJsonSchema({
+			type: "array",
+			items: [{ type: "string" }, { type: "number" }],
+			additionalItems: { type: "boolean" }
+		})
+		attest(tItemsVariadic.json).snap({
+			minLength: 2,
+			proto: "Array",
+			sequence: {
+				prefix: ["string", "number"],
+				variadic: [{ unit: false }, { unit: true }]
+			}
+		})
+		attest(tItemsVariadic.allows(["foo", 1])).equals(true)
+		attest(tItemsVariadic.allows([1, "foo", true])).equals(false)
+		attest(tItemsVariadic.allows([false, "foo", 1])).equals(false)
+		attest(tItemsVariadic.allows(["foo", 1, true])).equals(true)
+	})
+
+	it("contains", () => {
+		const tContains = parseJsonSchema({
+			type: "array",
+			contains: { type: "number" }
+		})
+		const predicateRef =
+			tContains.internal.firstReferenceOfKindOrThrow(
```

Comment:
You can replace these with the named function strategy to make the tests a bit less verbose

Reply:
Done in dcd08139